js/style/style.js

```diff
@@ -68,10 +76,32 @@ function Style(stylesheet, animationLoop) {
         this.fire('load');
     }.bind(this);
 
-    if (typeof stylesheet === 'string') {
-        ajax.getJSON(normalizeURL(stylesheet), loaded);
+    var sourceTypesLoaded = function(err) {
```

Comment:
@jfirebaugh Yeah, once this refactor settles, I think this async weirdness can be eliminated.  It's here now only as a quick and dirty way to validate/show that `addSourceType` works, but ultimately there's no reason that core source types need to be added asynchronously through the `addSourceType` interface.

I do think `addSourceType` should:
 - (a) be a public, non-underscored method -- because its purpose is to allow third-party source types can be plugged in, which is the motivating reason for this PR; and
 - (b) be async -- because if a source type needs to provide a WorkerSource implementation, then the process of having all the workers load that implementation via `importScript` is async.